### PR TITLE
Fix up Makefile, to only build when needed

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Dan Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Broken this out from spec pull request https://github.com/kubernetes-incubator/cri-o/pull/136

**Comment by rhatdan** [comment]
@cyphar @runcom @mrunalp PTAL

File: Makefile [hunk 1]
```diff
@@ -38,24 +40,24 @@ conmon:
...
 	make -C $@
 
-ocid: ${OCID_LINK}
-	go build --tags "$(BUILDTAGS)" -o ocid ./cmd/server/
+ocid: $(wildcard oci/*.go) $(wildcard cmd/server/*.go) $(wildcard server/*.go) ${OCID_LINK}
+	go build --tags "$(BUILDTAGS)" -o $@ ./cmd/server/
 
-ocic: ${OCID_LINK}
-	go build -o ocic ./cmd/client/
+ocic: $(wildcard oci/*.go) $(wildcard cmd/server/*.go) $(wildcard server/*.go)
```
Comment (cyphar) [inline Makefile]: `$(shell find . -regexp '\.go$')` should work. But why have we dropped `OCID_LINK`?

Comment (runcom) [inline Makefile]: We need OCID_LINK in order to build the binaries with the correct vendored dependencies

**Comment by rhatdan** [comment]
OCID_LINK is on ocid,  Putting it on both ocid and ocic causes a look which causes both to be rebuilt. on every make binaries.

**Comment by rhatdan** [comment]
Switched to find . -name \*.go

**Comment by mrunalp** [comment]
LGTM

**Comment by cyphar** [comment]
@rhatdan You can use this to make it so that it doesn't get rebuilt each time:

```
ocic: $(sources) | $(OCID_LINK)
ocid: $(sources) | $(OCID_LINK)
```

It's a GNU Make-ism. I'll open a PR to do it (the issue with the current setup is that `make ocic` will fail on newly cloned repos).

**Comment by rhatdan** [comment]
cyphar Currently ocic does not need OCID_LINK, but if this future proofs it, all the better.

**Comment by cyphar** [comment]
Doesn't it? Aren't we using `GOPATH=` for both binaries?

**Comment by runcom** [comment]
We are afaict

**Comment by rhatdan** [comment]
The ocic is not pulling in any OCID libraries.  
You can do make clean
make ocic 

and it builds.

**Comment by runcom** [comment]
@rhatdan I believe what @cyphar meant is that we still need to build `ocic` with the dependencies bundled in this repository under `vendor/src` rather than relying on the host's GOPATH.

**Pull request merged**